src/core/unit.c

```diff
@@ -5394,29 +5396,31 @@ int unit_prepare_exec(Unit *u) {
         return 0;
 }
 
-static void log_leftover(pid_t pid, int sig, void *userdata) {
+static bool log_leftover(pid_t pid, int sig, void *userdata) {
         _cleanup_free_ char *comm = NULL;
 
         (void) get_process_comm(pid, &comm);
 
         if (comm && comm[0] == '(') /* Most likely our own helper process (PAM?), ignore */
-                return;
+                return false;
 
-        log_unit_warning(userdata,
+        log_unit_warning((Unit *) userdata,
```

Comment:
the explicit cast to Unit* is unnecessary, as in C "void*" casts implicitly to any type